app/forms/bulk_submission_form.rb

```diff
@@ -78,4 +92,95 @@ def file_size(file)
     return 0 if file.nil?
     File.size(file&.tempfile)
   end
+
+  def csv
+    @csv ||= CSV.parse(File.read(uploaded_file), headers: true)
+  rescue StandardError
+    nil
```

Comment:
if an error occurs in parsing perhaps we should be adding a generic "unparsable" type error and surface this

Reply:
👍 have added unparseable file error message here (and test that it is being raised for your second example of a file with blank rows).